src/java.base/share/classes/java/lang/foreign/MemorySegment.java

```diff
@@ -55,65 +56,81 @@
 import static java.lang.foreign.ValueLayout.JAVA_BYTE;
 
 /**
- * A memory segment models a contiguous region of memory. A memory segment is associated with both spatial
- * and temporal bounds (e.g. a {@link MemorySession}). Spatial bounds ensure that memory access operations on a memory segment cannot affect a memory location
- * which falls <em>outside</em> the boundaries of the memory segment being accessed. Temporal bounds ensure that memory access
- * operations on a segment cannot occur after the memory session associated with a memory segment has been closed (see {@link MemorySession#close()}).
- *
- * There are many kinds of memory segments:
+ * A memory segment provides access to a contiguous region of memory.
+ * <p>
+ * There are two kinds of memory segments:
  * <ul>
- *     <li>{@linkplain MemorySegment#allocateNative(long, long) native memory segments}, backed by off-heap memory;</li>
- *     <li>{@linkplain FileChannel#map(FileChannel.MapMode, long, long, MemorySession) mapped memory segments}, obtained by mapping
- * a file into main memory ({@code mmap}); the contents of a mapped memory segments can be {@linkplain #force() persisted} and
- * {@linkplain #load() loaded} to and from the underlying memory-mapped file;</li>
- *     <li>{@linkplain MemorySegment#ofArray(int[]) array segments}, wrapping an existing, heap-allocated Java array; and</li>
- *     <li>{@linkplain MemorySegment#ofBuffer(Buffer) buffer segments}, wrapping an existing {@link Buffer} instance;
- * buffer memory segments might be backed by either off-heap memory or on-heap memory, depending on the characteristics of the
- * wrapped buffer instance. For instance, a buffer memory segment obtained from a byte buffer created with the
- * {@link ByteBuffer#allocateDirect(int)} method will be backed by off-heap memory.</li>
+ *     <li>A <em>heap segment</em> is backed by, and provides access to, a region of memory inside the Java heap (an "on-heap" region).</li>
+ *     <li>A <em>native segment</em> is backed by, and provides access to, a region of memory outside the Java heap (an "off-heap" region).</li>
  * </ul>
+ * Heap segments can be obtained by calling one of the {@link MemorySegment#ofArray(int[])} factory methods.
```

Comment:
There is just one MemorySegment#ofArray(int[]) method. Is there a way of expressing a link to the method reference `MemorySegment::ofArray` (which covers all the overloads)?

Reply:
not that I know of